src/bun.js/bindings/bindings.cpp

```diff
@@ -1556,6 +1556,38 @@ std::optional<bool> specialObjectsDequal(JSC::JSGlobalObject* globalObject, Mark
         return gp1->target()->m_globalThis == gp2->target()->m_globalThis;
     }
     default: {
```

Comment:
This should use `NumberObjectType` and `BooleanObjectType`. We should not be using these casts on every single value.